src/geop/geoprocessing.py

```diff
@@ -258,3 +268,47 @@ def statistics(geom, raster_path, stat):
         return layer.std()
     else:
         raise Exception("{0} has not been implemented".format(stat))
+
+
+def render_tile(geom, raster_path, user_palette):
+    """
+    Generates a visual PNG map tile from a vector polygon
+
+    Args:
+        geom (Shapely Geometry): A polygon corresponding to a TMS tile
+            request boundary
+
+        raster_path (string): A local file path to a raster in EPSG:3857
+            to generate visual tile from
+
+        uer_palette (optional list): A sequence of RGB triplets whose index
+            corresponds to the raster value which will be rendered. If
+            provided, will override a ColorTable defined in the raster
+    Returns:
+        Byte Array of image in the PNG format
+    """
+    tile, palette = tile_read(geom, raster_path)
+    return render_tile_from_data(tile, user_palette or palette)
+
+
+def render_tile_from_data(tile, palette):
+    """
+    Generates a visual PNG map tile from an ndarray of raster data
+
+    Args:
+        tile (ndarray) : A square 256x256 array of raster values
+
+        palette (list<int>): A list of RGB values to render `tile`
+
+    Returns:
+        Byte Array of image in the PNG format
+    """
+    img = Image.fromarray(tile, mode='P')
+
+    if len(palette):
+        img.putpalette(palette)
+
+    img_data = BytesIO()
+    img.save(img_data, 'png')
```

Comment:
Looks like the Django `Response` object can be written to like a file object, which would have allowed us to bypass this intermediate `BytesIO` object potentially. I haven't been able to find a comparable way to do that with Flask. Their process for streaming seems to be based around generators.

It also looks like `save()` here is the right way to do things. I was initially looking at `Image.tobytes`, but that doesn't seem to return compressed data.